sbi/neural_nets/density_estimators/flow.py

```diff
@@ -52,12 +65,60 @@ def sample(self, sample_shape: Shape, condition: Tensor) -> Tensor:
         Returns:
             Samples.
         """
+        self._check_for_invalid_condition_shape(condition)
+
+        num_samples = torch.Size(sample_shape).numel()
+        x_dim = len(self._x_shape)
+
+        if len(condition.shape) == x_dim:
+            # nflows.sample() expects conditions to be batched.
+            condition = condition.unsqueeze(0)
+            samples = self.net.sample(num_samples, context=condition).reshape(
+                (*sample_shape, -1)
+            )
+        else:
+            # For batched conditions, we need to reshape the conditions and the samples
+            batch_dims = condition.shape[:-x_dim]
+            condition = condition.reshape(-1, *self._x_shape)
+            samples = self.net.sample(num_samples, context=condition).reshape(
+                (*batch_dims, *sample_shape, -1)
+            )
+
+        return samples
+
+    def sample_and_log_prob(
+        self, sample_shape: torch.Size, condition: Tensor, **kwargs
+    ) -> Tuple[Tensor, Tensor]:
+        r"""Return samples and their density from the density estimator.
+
+        Args:
+            sample_shape (torch.Size): Shape of the samples to return.
```

Comment:
no need for type hints in the docstring if we give them above. will be filled in by `mkdcos`.